src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/domain/move/MoveRepository.kt

```diff
@@ -1,11 +1,31 @@
 package uk.gov.justice.digital.hmpps.pecs.jpc.domain.move
 
 import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Query
+import org.springframework.data.repository.query.Param
 import uk.gov.justice.digital.hmpps.pecs.jpc.domain.price.Supplier
 
 interface MoveRepository : JpaRepository<Move, String> {
 
   fun findAllByMoveIdIn(ids: List<String>): List<Move>
 
   fun findByReferenceAndSupplier(ref: String, supplier: Supplier): Move?
+
+  @Query(
+    """
+    select m
+      from Move m
+     where m.supplier = :supplier
+       and m.moveYear = :year
+       and m.moveMonth = :month
+       and m.status= 'completed'
+       and (m.moveType is null or m.dropOffOrCancelledDateTime is null)
```

Comment:
if a move has been classified as completed but either the move type cannot be established or the drop of date is null they are not included in the overall pricing therefore not visible to the end user.